### PR TITLE
[do not merge] [TERRA-278] AWS credentials support in CLI

### PR DESCRIPTION
**Pull request by dexamundsen**
draft changes for AWS credentials support in CLI

Assumptions:
- Resources will override this method as needed
- console link is fetched along with credentials. If standalone support for console link is needed, we can implement a similar `terra resource console-link`
- return fields (i.e. credential object fields) may not be the same across resources. If they are expected to be the same, a businessObject record can be created for ResourceCredentials

expected use on a AWS resource

```
> terra resource credentials --name=s3-name-1 --format=json
{
  "AccessKeyId": "x",
  "SecretAccessKey": "x",
  "SessionToken": "x",
  "Expiration": "x"
}

> terra resource credentials --name=s3-name-1 --format=json --console-link
{
  "AccessKeyId": "x",
  "SecretAccessKey": "x",
  "SessionToken": "x",
  "Expiration": "x",
  "ConsoleLink": "x"
}

> terra resource credentials --name=gcs-bucket-1
Error: Credentials not supported for resource type: GCS_BUCKET
```

### CODE REVIEW

File: src/main/java/bio/terra/cli/businessobject/resource/AwsS3StorageFolder.java [hunk 1]
```diff
@@ -121,6 +124,34 @@ public static String resolve(
...
+            .getAwsS3StorageFolderCredential(
+                Context.requireWorkspace().getUuid(),
+                id,
+                scope == CredentialsAccessScope.READ_ONLY
+                    ? AwsCredentialAccessScope.READ_ONLY
+                    : AwsCredentialAccessScope.WRITE_READ,
+                duration);
+
+    JSONObject object = new JSONObject();
+    object.put("AccessKeyId", awsCredential.getAccessKeyId());
```
Comment (dexamundsen) [inline src/main/java/bio/terra/cli/businessobject/resource/AwsS3StorageFolder.java:141]: Add version

**Review by jmczerk — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/resource/AwsS3StorageFolder.java (hunk 1 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/businessobject/resource/AwsS3StorageFolder.java:141]: Hard coded to 1 in WSM schema object https://github.com/DataBiosphere/terra-workspace-manager/pull/1215
this keeps CLI and wsm responses consistent (and same)

File: src/main/java/bio/terra/cli/command/resource/Credentials.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      names = "--scope",
+      description = "Set the credentials access scope: ${COMPLETION-CANDIDATES}.",
+      defaultValue = "READ_ONLY",
+      showDefaultValue = CommandLine.Help.Visibility.ALWAYS)
+  private Resource.CredentialsAccessScope scope;
+
+  @CommandLine.Option(
+      names = "--duration",
+      defaultValue = "900",
+      description = "Duration of access (in seconds), minimum: 900s, maximum: 1800s")
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/command/resource/Credentials.java]: fixed

**Comment by dexamundsen** [comment]
tests and complete review handled in https://github.com/DataBiosphere/terra-cli/pull/459

**Pull request merged**